src/index.js

```diff
@@ -26,10 +29,21 @@ module.exports = postcss.plugin('postcss-modules', (opts = {}) => {
   }
 
   return (css, result) => {
+    const inputFile = css.source.input.file;
+
     const resultPlugins = result.processor.plugins
       .filter(plugin => plugin.postcssPlugin !== 'postcss-modules');
 
-    const plugins = [...defaultPlugins[scopeBehaviour], ...resultPlugins];
+    let pluginList = defaultPlugins[defaultScopeBehaviour];
+
+    if (globalModulePathWhitelist) {
+      const isGlobalModule = globalModulePathWhitelist
+        .filter(regex => inputFile.match(regex))
+        .length !== 0;
+      pluginList = defaultPlugins[isGlobalModule ? 'global' : 'local'];
```

Comment:
`opts.scopeBehaviour = 'global'` can be overwritten here. It's better to rewrite this string as
`if (isGlobalModule) pluginList = defaultPlugins['global']`